packages/template-retail-react-app/app/hooks/use-update-shopper-context.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useEffect} from 'react'
+import {useLocation} from 'react-router-dom'
+import {
+    useUsid,
+    useShopperContext,
+    useShopperContextsMutation
+} from '@salesforce/commerce-sdk-react'
+import {isServer, isHydrated} from '@salesforce/retail-react-app/app/utils/utils'
+import {useQueryClient} from '@tanstack/react-query'
+
+// Hooks
+import useMultiSite from '@salesforce/retail-react-app/app/hooks/use-multi-site'
+
+// Constants
+import {useShopperContextSearchParams} from '@salesforce/retail-react-app/app/hooks/use-shopper-context-search-params'
+
+/*
+ * This hook will set the shopper context when search params pertinant
+ * to shopper context are present.
+ */
+export const useUpdateShopperContext = () => {
```

Comment:
created a separate hook for handling creating/updating shopper context. In the future if we add more ways for updating context (e.g., button) it can be called from within here